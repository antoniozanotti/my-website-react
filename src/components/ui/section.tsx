import React from "react";
import { ReactNode } from "react";
import Heading from "@/components/ui/heading";

export interface SectionProps extends React.ComponentProps<"section"> {
  title: string;
  children?: ReactNode;
}

export default function Section({
  title,
  children,
  ...props
}: SectionProps) {
  return (
    <section {...props}>
      <Heading level="h2" title={title} />
      {children}
    </section>
  );
}
