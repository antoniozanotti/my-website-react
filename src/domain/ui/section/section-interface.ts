export interface SectionInterface extends React.ComponentProps<"section"> {
  title: string;
  anchor?: string;
}