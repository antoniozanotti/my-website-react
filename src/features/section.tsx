import React from "react";
import { ReactNode } from "react";
import Heading from "@features/heading";

export interface SectionProps {
  title: string;
  className?: string;
  children?: ReactNode;
}

export default function Section({
  title,
  children,
  className = "",
}: SectionProps) {
  return (
    <section className={className}>
      <Heading level="h2" title={title} />
      {children}
    </section>
  );
}
