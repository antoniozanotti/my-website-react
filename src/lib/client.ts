import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";

const TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
const SPACE = process.env.CONTENTFUL_SPACE_ID;
const ENVIRONMENT = process.env.CONTENTFUL_ENVIRONMENT_ID;
const URL = `https://graphql.contentful.com/content/v1/spaces/${SPACE}/environments/${ENVIRONMENT}`;

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: URL,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
      },
    }),
  });
});
