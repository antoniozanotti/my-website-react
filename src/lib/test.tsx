import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
