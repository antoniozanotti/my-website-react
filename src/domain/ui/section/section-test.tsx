import { test, expect } from "@playwright/experimental-ct-react";
import { Section } from "./section";

test.describe("Section", () => {
  test("should render section", async ({ mount }) => {
    const component = await mount(<Section title="Test Title" />);
    await expect(component.locator(":scope:is(section)")).toHaveCount(1);
  });

  test("should accept anchor", async ({ mount }) => {
    const component = await mount(<Section  title="Test Title" anchor="TestId" />);
    await expect(component.locator("h2")).toHaveId("TestId");
  });

  test("should accept className", async ({ mount }) => {
    const component = await mount(<Section  title="Test Title" className='test' />);
    await expect(component).toHaveClass(/test$/);
  });

  test("should render children", async ({ mount }) => {
    const component = await mount(<Section title="Test Title"><span>Children</span></Section>);
    await expect(component.locator("span")).toHaveText("Children");
  });
});
