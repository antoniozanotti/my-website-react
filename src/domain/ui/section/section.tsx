import { Heading } from "../heading/heading";
import { SectionInterface } from "./section-interface";

export function Section({
  title,
  anchor,
  children,
  ...props
}: SectionInterface) {
  return (
    <section {...props}>
      <Heading level="h2" title={title} anchor={anchor} />
      {children}
    </section>
  );
}
