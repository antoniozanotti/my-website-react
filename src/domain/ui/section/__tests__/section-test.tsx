import { screen, render, firstComponent } from "@/lib/test";
import { Section } from "../section";

describe("Section", () => {
  it("should render section", async () => {
    const { container } = render(<Section title="Test Title" />);
    const component = container.querySelector("section");

    expect(component).toBeVisible();
    expect(container).toMatchSnapshot();
  });

  it("should accept anchor", async () => {
    const { container } = render(<Section title="Test Title" anchor="TestId" />);
    const component = await screen.findByRole("heading", { level: 2 });

    expect(component).toHaveAttribute("id", "TestId");
    expect(container).toMatchSnapshot();
  });

  it("should accept className", async () => {
    const { container } = render(<Section title="Test Title" className="test" />);
    const component = firstComponent(screen);

    expect(component).toHaveClass(/test$/);
    expect(container).toMatchSnapshot();
  });

  it("should render children", async () => {
    const { container } = render(
      <Section title="Test Title">
        <span>Children</span>
      </Section>
    );

    expect(container).toHaveTextContent("Children");
    expect(container).toMatchSnapshot();
  });
});
