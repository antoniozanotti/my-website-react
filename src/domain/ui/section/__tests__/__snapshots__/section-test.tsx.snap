// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should accept anchor 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <section>
          <h2
            class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent-1 dark:text-dark-accent-1 drop-shadow mt-[164px] "
            id="TestId"
          >
            Test Title
          </h2>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`Section should accept className 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <section
          class="test"
        >
          <h2
            class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent-1 dark:text-dark-accent-1 drop-shadow mt-[164px] "
            id=""
          >
            Test Title
          </h2>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`Section should render children 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <section>
          <h2
            class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent-1 dark:text-dark-accent-1 drop-shadow mt-[164px] "
            id=""
          >
            Test Title
          </h2>
          <span>
            Children
          </span>
        </section>
      </div>
    </div>
  </div>
</div>
`;

exports[`Section should render section 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <section>
          <h2
            class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent-1 dark:text-dark-accent-1 drop-shadow mt-[164px] "
            id=""
          >
            Test Title
          </h2>
        </section>
      </div>
    </div>
  </div>
</div>
`;
